scripts/compatibility_matrix.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+
+# License: CC0 https://creativecommons.org/share-your-work/public-domain/cc0/
+
+# A messy script that figures out ZFS features. It's very messy, sorry. I am
+# not responsible if this script eats your laundry.
+#
+# This uses manpages, because I'm lazy. If your manpages are wrong, you have a
+# bug.
+#
+# If the script is wrong, or could be improved, feel free to contact me on
+# freenode, and tell me why it's wrong. My nick is zgrep.
+#
+# 2018-07-05: Created.
+# ????-??-??: Many things happened.
+# 2020-02-23: Applied patch by rlaager.
+# 2020-02-30: Show domain prefixes (via Vlad Bokov), partially apply patch by
+#   rlaager.
+# 2020-03-05: Patch by rlaager (allocation_classes, ZoL -> openzfs).
+# 2020-10-07: Rework by gmelikov for openzfs-docs
+
+import logging
+import sys
+
+from collections import defaultdict
+from urllib.request import urlopen
+from datetime import datetime
+from re import sub as regex, findall
+from json import loads as dejson
+
+LOG = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+
+if len(sys.argv) == 1:
+    path = '.'
+elif len(sys.argv) != 2:
+    print('Usage:', sys.argv[0], 'path')
+    exit(1)
+else:
+    path = sys.argv[1]
+
+
+def openzfs():
+    sources = {'master': 'https://raw.githubusercontent.com/openzfs/zfs/'
+               'master/man/man5/zpool-features.5'}
+    # TODO(gmelikov): use git tags from OpenZFS repo
+    with urlopen('https://zfsonlinux.org') as web:
+        versions = findall(r'download/zfs-([0-9.]+)',
+                           web.read().decode('utf-8', 'ignore'))
+    for ver in set(versions):
+        sources[ver] = ('https://raw.githubusercontent.com/openzfs/zfs/'
+                        'zfs-{}/man/man5/zpool-features.5'.format(ver))
+    return sources
+
+
+def openzfsonosx():
+    sources = {'master': 'https://raw.githubusercontent.com/openzfsonosx/'
+               'zfs/master/man/man5/zpool-features.5'}
+    with urlopen('https://api.github.com/repos/openzfsonosx/zfs/tags') as web:
+        try:
+            tags = dejson(web.read().decode('utf-8', 'ignore'))
+            tags = [x['name'].lstrip('zfs-') for x in tags]
+            tags.sort()
+            latest = tags[-1]
+            tags = [tag for tag in tags if 'rc' not in tag]
+            if 'rc' not in latest:
+                tags = tags[-3:]
+            else:
+                tags = tags[-2:] + [latest]
+        except Exception:
+            tags = []
+    for ver in tags:
+        sources[ver] = ('https://raw.githubusercontent.com/openzfsonosx/zfs/'
+                        'zfs-{}/man/man5/zpool-features.5'.format(ver))
+    return sources
+
+
+def freebsd_pre_openzfs():
+    # TODO(gmelikov): add FreeBSD HEAD (OpenZFS version)?
```

Comment:
I think this is what you're getting at, but if not... FreeBSD is still merging the OpenZFS code into their repository, right? So there could be some lag between OpenZFS upstream and the FreeBSD tree? I believe the goal is that there are no differences (certainly at a feature level), but that doesn't mean there wouldn't be delay. So it may still make sense to get the features from their version, to account for that lag.

Reply:
You've got it right, clarified it!